src/main/java/yasp/Main.java

```diff
@@ -2,35 +2,61 @@
 
 import java.io.IOException;
 import java.io.InputStream;
+import java.io.FileInputStream;
 import java.io.OutputStream;
 import java.net.InetSocketAddress;
 import com.sun.net.httpserver.HttpExchange;
 import com.sun.net.httpserver.HttpHandler;
 import com.sun.net.httpserver.HttpServer;
     
 public class Main {
+    
+    public static void main(String[] args) {
+	try { 
+                       
+	    if (args.length > 0 && args[0].equals("--file")) { System.exit(parse(args[1])); }
```

Comment:
To be honest, if you want to include the Main class, I'd rather have a proper arg-parsing, like Apache Commons CLI or similar.